### ISSUE TITLE
Characters are not handled correctly in keybindings

### ISSUE THREAD

**Issue by ClementNerma** [nushell/reedline#546]
I'm trying to integrate custom keybindings with the editor, here is what I got:

```rust
let mut keybindings = default_emacs_keybindings();

keybindings.add_binding(
    KeyModifiers::NONE,
    KeyCode::Char('c'),
    ReedlineEvent::Edit(vec![EditCommand::Clear]),
);

// ...
rl.with_edit_mode(Emacs::new(keybindings)) // ...
```

This doesn't work. If I type the letter "c" nothing happens. I also tried with various keys, tried in various keyboard layouts including QWERTY (which is not my native layout), nothing works.

If I use another key code such as `KeyCode::Tab`  it works perfectly. But with custom characters specifically it doesn't seem to work.

Any idea why?

**Comment by ClementNerma** [comment]
So I did a bit of testing, and here are some of the results:

`Ctrl + Backspace` produces a `Ctrl + W`
`Shift + Enter` produces an `Enter` event (no modifier)

**Comment by sholderbach** [comment]
The binding you describe in the initial description should be possible after #580 

For the other problems you describe `cargo run --example event_listener` can be helpful to see if we are actually able to receive or disambiguate the event

> `Ctrl + Backspace` produces a `Ctrl + W` 

I think this combination can be an artifact of both how the terminal handles it and it is encoded in the basic ASCII representation underlying the ANSI terminal protocol. e.g. on Windows Terminal in WSL2 I can observe `Ctrl-H` instead of `Ctrl-Backspace` with `cargo run --example event_listener`/nu's `keybinding listen` as this corresponds to the character that aligns in the [Control code block of ASCII](https://en.wikipedia.org/wiki/ASCII#Control_code_chart)

> `Shift + Enter` produces an `Enter` event (no modifier)

This is also very terminal emulator implementation dependent.

**Comment by ClementNerma** [comment]
Hi, I'm stuck again on this problem after coming back to a project of mine that uses reedline.

How do you work around this limitation? `cargo run --example listener` gives me the correct keys, but `ReedlineRawEvent::into()` doesn't.

**Comment by sholderbach** [comment]
Can you open a new issue with the particular difference you observe between the listener and the internals?

**Comment by ClementNerma** [comment]
Followed at #663
